### PR TITLE
Update models to enable save/load

### PR DESCRIPTION
**Pull request by Derek-Wds**
## Description

This is related to the issues: [#211](https://github.com/microsoft/qlib/issues/211) and [#225](https://github.com/microsoft/qlib/issues/225). Users can directly save models through `R.save_objects`.

## How Has This Been Tested?
- [ ] Pass the test by running: `pytest qlib/tests/test_all_pipeline.py` under upper directory of `qlib`.
- [ ] If you are adding a new feature, test on your own test scripts.

## Screenshots of Test Results (if appropriate):
1. Pipeline test:
2. Your own tests:

## Types of changes

- [x] Fix bugs
- [x] Add new feature
- [ ] Update documentation

### CODE REVIEW

File: qlib/data/data.py [hunk 1]
```diff
@@ -478,13 +478,13 @@ def expression_calculator(inst, start_time, end_time, freq, column_names, spans=
 
         data = pd.DataFrame(obj)
         _calendar = Cal.calendar(freq=freq)
-        data.index = _calendar[data.index.values.astype(np.int)]
+        data.index = _calendar[data.index.values.astype(int)]
```
Comment (you-n-g) [inline qlib/data/data.py:481]: Why is this necessary ?

Comment (Derek-Wds) [reply qlib/data/data.py:481]: `np.int`, and `np.bool` is deprecated in numpy 1.20.0. We can use `int` directly or specify `np.int32` or `np.int64`.
